Description: Adjust the source/target level automatically for building with recent JDKs
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/subprojects/language-jvm/src/main/java/org/gradle/api/internal/tasks/compile/DefaultJvmLanguageCompileSpec.java
+++ b/subprojects/language-jvm/src/main/java/org/gradle/api/internal/tasks/compile/DefaultJvmLanguageCompileSpec.java
@@ -32,6 +32,13 @@
     private String sourceCompatibility;
     private String targetCompatibility;
 
+    private List<String> unsupportedLanguageLevels = java.util.Arrays.asList(new String[]{"1.1", "1.2", "1.3", "1.4", "1.5", "5", "1.6", "6", "1.7", "7"});
+
+    /** Detect if a Debian build is in process */
+    static boolean isDebianBuild() {
+        return System.getenv("DEB_BUILD_ARCH") != null;
+    }
+
     @Override
     public File getWorkingDir() {
         return workingDir;
@@ -100,6 +107,11 @@
 
     @Override
     public void setSourceCompatibility(String sourceCompatibility) {
+        if (isDebianBuild() && unsupportedLanguageLevels.contains(sourceCompatibility)) {
+            System.err.println( "Use of source " + sourceCompatibility + " is no longer supported, switching to 8" );
+            sourceCompatibility = "8";
+        }
+
         this.sourceCompatibility = sourceCompatibility;
     }
 
@@ -110,6 +122,11 @@
 
     @Override
     public void setTargetCompatibility(String targetCompatibility) {
+        if (isDebianBuild() && unsupportedLanguageLevels.contains(targetCompatibility)) {
+            System.err.println( "Use of target " + targetCompatibility + " is no longer supported, switching to 8" );
+            targetCompatibility = "8";
+        }
+
         this.targetCompatibility = targetCompatibility;
     }
 }
